data/trie/branchNode.go

```diff
@@ -496,3 +496,36 @@ func (bn *branchNode) print(writer io.Writer, index int) {
 		child.print(writer, index+len(str)-1+len(str2))
 	}
 }
+
+func (bn *branchNode) deepClone() node {
```

Comment:
You should check if bn.isEmptyOrNil(), as all the other methods do.

Reply:
ok, will add checks. The rest of our project lacks these kind of tests.